virtool/http/policy.py

```diff
@@ -46,18 +53,39 @@ async def check(self, req, handler, client: AbstractClient):
             raise HTTPForbidden(text="Requires administrative privilege")
 
 
-class PermissionsRoutePolicy(DefaultRoutePolicy):
-    def __init__(self, object_type, object_id, permission: PermissionType):
+class PermissionRoutePolicy(DefaultRoutePolicy):
+    def __init__(
+        self,
+        object_type: ResourceType,
+        object_id: Union[int, str],
+        permission: PermissionType,
+    ):
         self.object_type = object_type
         self.object_id = object_id
         self.permission = permission
 
-    async def check(self, req, handler, client):
-        abs_client = req.app["auth"]
-        if not client.administrator and not await abs_client.check(
+    async def check(self, req: Request, handler: Callable, client):
+        """
+        Checks if the client has the required permission for the object.
+
+        Raises ``HTTPForbidden`` if the client does not have the required permission.
+
+        The check will pass if:
+        * The user is an administrator.
+        * The user has the required permission in their legacy MongoDB-based
+          permissions.
+        * The permission check passes against the authorization client.
+
+        """
+        if client.administrator or self.permission.value in client.permissions:
```

Comment:
```suggestion
        if client.administrator or self.permission.value.id in client.permissions:
```
Permission.value.id here since using the new Permission enum.

Reply:
```suggestion
        if client.administrator or client.permissions[self.permission.value.id]:
```
Previously, the MongoDB legacy check would pass regardless if the user had the permission or not.